src/gentropy/datasource/ukb_ppp_eur/summary_stats.py

```diff
@@ -0,0 +1,162 @@
+"""Summary statistics ingestion for UKB PPP (EUR)."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import pyspark.sql.functions as f
+import pyspark.sql.types as t
+from pyspark.sql import SparkSession
+
+from gentropy.common.spark_helpers import neglog_pvalue_to_mantissa_and_exponent
+from gentropy.dataset.summary_statistics import SummaryStatistics
+
+
+@dataclass
+class UkbPppEurSummaryStats:
+    """Summary statistics dataset for UKB PPP (EUR)."""
+
+    @classmethod
+    def from_source(
+        cls: type[UkbPppEurSummaryStats],
+        spark: SparkSession,
+        raw_summary_stats_path: str,
+        tmp_variant_annotation_path: str,
+        chromosome: str,
+    ) -> SummaryStatistics:
+        """Ingests all summary stats for UKB PPP (EUR) data.
+
+        Args:
+            spark (SparkSession): Spark session object.
+            raw_summary_stats_path (str): Input raw summary stats path.
+            tmp_variant_annotation_path (str): Input variant annotation dataset path.
+            chromosome (str): Which chromosome to process.
+
+        Returns:
+            SummaryStatistics: Processed summary statistics dataset for a given chromosome.
+        """
+        # Read the precomputed variant annotation dataset.
+        va_df = (
```

Comment:
I don't think so: this is not a "true" VariantAnnotation dataset; at this point it already has some additional fields precomputed and appended to it, so it wouldn't pass the schema check. This is done in order to speed up calculations during the merge step, as it would be otherwise quite expensive computationally